frontend/components/SignUpForNotifications.js

```diff
@@ -36,6 +36,7 @@ class SignUpForNotifications extends React.Component {
     this.facebookScopeRef = null;
     this.onSubscribeToggleChange = this.onSubscribeToggleChange.bind(this);
     this.closeModal = this.closeModal.bind(this);
+    this.handleClick = this.props.handleClick;
```

Comment:
Lets not do this here - props can change when the component is updated (componentDidUpdate) - so saving this in the constructor would ignore any future updates of this prop. Lets just do this.props.handleClick throughout the component!